packages/framework/fluid-static/src/utils.ts

```diff
@@ -31,16 +33,25 @@ export const isSharedObjectClass = (obj: any): obj is SharedObjectClass<any> =>
  * of both SharedObject or DataObject. This function seperates the two and returns a registery
  * of DataObject types and an array of SharedObjects.
  */
-export const parseDataObjectsFromSharedObjects = (schema: ContainerSchema):
-    [NamedFluidDataStoreRegistryEntry[], IChannelFactory[]] => {
+export const parseDataObjectsFromSharedObjects = (
+    schema: ContainerSchema,
+): [NamedFluidDataStoreRegistryEntry[], IChannelFactory[]] => {
     const registryEntries: Set<NamedFluidDataStoreRegistryEntry> = new Set();
     const sharedObjects: Set<IChannelFactory> = new Set();
+    const usedDataObjTypes: Set<string> = new Set();
 
     const tryAddObject = (obj: LoadableObjectClass<any>) => {
         if (isSharedObjectClass(obj)) {
             sharedObjects.add(obj.getFactory());
```

Comment:
@ChumpChief maybe worth check next week if we should de-dep sharedObjects as well.

Reply:
Alternatively, maybe let the Set do the deduping prior to the iteration below on 50-58, which should cover both --

```ts
// Add the object types that will be initialized
const initialObjectTypes = Object.values(schema.initialObjects);
const dynamicObjectTypes = schema.dynamicObjectTypes ?? [];
const dedupedObjects = new Set([...initialObjectTypes, ...dynamicObjectTypes]);
dedupedObjects.forEach(tryAddObject);
```

Reply:
Ok. Looking at usage, it should be safe to dedup sharedObjects as well. Made the change for the full dedup.